CMakeLists.txt

```diff
@@ -19,7 +19,7 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}"
 option(ICUB_COMPILE_BINDINGS "Compile optional language bindings"  FALSE)
 
 set(YCM_REQUIRED_VERSION 0.15.3)
-set(YARP_REQUIRED_VERSION 3.8.1)
+set(YARP_REQUIRED_VERSION 3.9.0)
```

Comment:
We need to wait until `yarp-3.9.0` is shipped. Our CI does not rely on `yarp@master`.
cc @randaz81 @traversaro @sgiraz